src/scripts/Selector/SelectorTable.js

```diff
@@ -26,48 +26,49 @@ export default class SelectorTable extends Selector {
 		return false;
 	}
 
-	getTableHeaderColumns($table) {
-		let columns = {};
+	getTableHeaderColumns($table, verticalTable) {
 		let headerRowSelector = this.getTableHeaderRowSelector();
 		let $headerRow = $table.find(headerRowSelector);
-
-		if ($headerRow.length > 0) {
-			if ($headerRow.length > 1) {
-				if ($headerRow[0].nodeName === 'TR') {
-					$headerRow = $headerRow.find('th:first-child');
-					if ($headerRow.length === 0) {
-						console.log('%c Please specify row header cell selector ', 'background: red; color: white;');
+		let columns = {};
+		if (verticalTable) {
+			if ($headerRow.length > 0) {
+				if ($headerRow.length > 1) {
+					if ($headerRow[0].nodeName === 'TR') {
+						$headerRow = $headerRow.find('th:first-child');
+						if ($headerRow.length === 0) {
+							console.log('%c Please specify row header cell selector ', 'background: red; color: white;');
+						}
 					}
+				} else if ($headerRow.find('th').length) {
+					$headerRow = $headerRow.find('th');
+				} else if ($headerRow.find('td').length) {
+					$headerRow = $headerRow.find('td');
 				}
-			} else if ($headerRow.find('th').length) {
-				$headerRow = $headerRow.find('th');
-			} else if ($headerRow.find('td').length) {
-				$headerRow = $headerRow.find('td');
+				$headerRow.each(
+					function(i, value) {
+						let header = $(value)
+							.text()
+							.trim();
+						columns[header] = {
+							index: i + 1,
+						};
+					}.bind(this)
+				);
+				this.addMissingColumns($headerRow);
 			}
-
-			$headerRow.each(
-				function(i, value) {
-					let header = $(value)
-						.text()
-						.trim();
-					columns[header] = {
-						index: i + 1,
-					};
-				}.bind(this)
-			);
-
-			this.addMissingColumns($headerRow);
+		} else {
+			let limit = $headerRow[0].cells.length;
+			columns = SelectorTable.columnsMaker($headerRow, columns, 0, 0, '', 0, limit);
 		}
+
 		return columns;
 	}
 
 	addMissingColumns(headerRow) {
 		headerRow.each(
 			function(i, value) {
 				if (this.tableAddMissingColumns) {
-					let header = $(value)
-						.text()
-						.trim();
+					let header = $(value).text();
 					let column = $.grep(this.columns, function(h) {
```

Comment:
getDataColumnName here?

Reply:
I don't think so, this is not my code. Should I change it?

Reply:
We could not divide code on ours and not ours. Our goal is to make it better and understandable.